tests/runtests.rs

```diff
@@ -68,26 +54,69 @@ fn get_sysroot() -> PathBuf {
     PathBuf::from(sysroot.trim())
 }
 
-fn run_prog(path: &Path) -> i32 {
-    let status = Command::new("target/debug/mirkat")
-                    .arg(path)
-                    .status()
-                    .expect("mirkat bin not found");
-    status.code().unwrap()
+/// Invokes Mirkat when given a path to a valid Rust program. Compares the
+/// expected return code against that returned from Mirkat. This is surprisingly
+/// effective at catching bugs in Mirkat. It's not uncommon that when things go
+/// wrong, Mirkat blows up catastrophically.
+fn test_prog(path: &Path, expected_code: i32) -> Result<(), Vec<u8>> {
+    let out = run_prog(path);
+
+    match out.status.code().unwrap() {
```

Comment:
Wouldn't this be clearer as a traditional `if`?

Reply:
Yes it would, I did adjust this later to fix this. When I squash I will merge these into a single commit